src/thunderbots/software/radio_communication/main.cpp

```diff
@@ -1,61 +1,37 @@
 #include <ros/ros.h>
 #include <thunderbots_msgs/Primitive.h>
 #include <thunderbots_msgs/PrimitiveArray.h>
+#include <thunderbots_msgs/World.h>
 
 #include "ai/primitive/primitive.h"
 #include "ai/primitive/primitive_factory.h"
-#include "geom/point.h"
-#include "grsim_communication/grsim_backend.h"
 #include "util/constants.h"
 #include "util/logger/init.h"
 
-// Member variables we need to maintain state
-// They are kept in an anonymous namespace so they are not accessible outside this
-// file and are not created as global static variables.
-namespace
-{
-    // A vector of primitives. It is cleared each tick, populated by the callbacks
-    // that receive primitive commands, and is processed by the backend to send
-    // the Primitives to the robots using the radio.
-    std::vector<std::unique_ptr<Primitive>> primitives;
-}  // namespace
-// Callbacks
 void primitiveUpdateCallback(const thunderbots_msgs::PrimitiveArray::ConstPtr& msg)
 {
+    std::vector<std::unique_ptr<Primitive>> primitives;
     thunderbots_msgs::PrimitiveArray prim_array_msg = *msg;
     for (const thunderbots_msgs::Primitive& prim_msg : prim_array_msg.primitives)
     {
         primitives.emplace_back(AI::Primitive::createPrimitiveFromROSMessage(prim_msg));
     }
 }
 
-
 int main(int argc, char** argv)
 {
     // Init ROS node
     ros::init(argc, argv, "radio_communication");
     ros::NodeHandle node_handle;
 
     // Create subscribers to topics we care about
-    ros::Subscriber prim_array_sub = node_handle.subscribe(
+    ros::Subscriber primitive_subscriber = node_handle.subscribe(
         Util::Constants::AI_PRIMITIVES_TOPIC, 1, primitiveUpdateCallback);
 
     // Initialize the logger
     Util::Logger::LoggerSingleton::initializeLogger(node_handle);
 
-    // Initialize variables
-    primitives = std::vector<std::unique_ptr<Primitive>>();
-
-    // Main loop
-    while (ros::ok())
-    {
-        // Clear all primitives each tick
-        primitives.clear();
-
-        // Spin once to let all necessary callbacks run
-        // The callbacks will populate the primitives vector
-        ros::spinOnce();
-    }
+    ros::spin();
```

Comment:
Quick comment on what `ros::spin()` does would be good (same goes for other instances of it in this PR)

Reply:
comments added